core/access_api.php

```diff
@@ -895,3 +887,64 @@ function access_parse_array( array $p_access ) {
 
 	return $t_access_level;
 }
+
+/**
+ * Return an access threshold for which a user is not affected by the 'limit_reporters'
+ * option. Also, if the option is disabled globally, or for the specific user/project,
+ * a threshold of ANYBODY will be returned, meaning that any user can bypass this limitation.
+ *
+ * @param integer $p_project_id   Project id, or null for current project
+ * @param integer $p_user_id      User id, or null for current user
+ * @return mixed	A threshold definition
+ */
+function access_threshold_reporter_unlimited_view( $p_project_id = null, $p_user_id = null ) {
+	# 'limit_reporters' option were only supported for ALL_PROJECTS
+	# use this option if set, otherwise, check the new option for threshold
+	$t_limit_reporters = config_get( 'limit_reporters', null, $p_user_id, $p_project_id );
+	if( ON != $t_limit_reporters ) {
+		$t_threshold_can_view = config_get( 'limit_reporter_unless_threshold', null, $p_user_id, $p_project_id );
+		return $t_threshold_can_view;
+	}
+
+	$t_user = null === $p_user_id ? auth_get_current_user_id() : $p_user_id;
+	$t_project = null === $p_project_id ? helper_get_current_project() : $p_project_id;
```

Comment:
Move these lines to the begin of the function and use `$t_user` and `$t_project` in first two calls of `config_get`

```php
	$t_user = null === $p_user_id ? auth_get_current_user_id() : $p_user_id;
	$t_project = null === $p_project_id ? helper_get_current_project() : $p_project_id;
```